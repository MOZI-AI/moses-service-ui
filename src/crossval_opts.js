import React from "react";
import { Form, Input, Button, Icon, Divider, Row, Col, Tooltip } from "antd";

export class CrossValidationOptions extends React.Component {
  isValid() {
    const fieldsError = this.props.form.getFieldsError();
    return !Object.keys(fieldsError).some(field => fieldsError[field]);
  }

  render() {
    const { getFieldDecorator } = this.props.form;

    return (
      this.props.show && <div className="crossValidationOptionsFormWrapper">
        <Form>
          <Row gutter={16}>
            <Col span={8}>
              <Form.Item
                label={(
                  <Tooltip title="Number of folds for cross-validation">
                    Folds&nbsp;
                  </Tooltip>
                )}
                wrapperCol={{ span: 24 }}>
                {getFieldDecorator("folds", {
                  initialValue: this.props.defaults.folds,
                  rules: [{ required: true, message: "Please input folds!" },
                  {
                    validator: (rule, value, callback) => {
                      if (value < 1) {
                        callback("Number of folds must be one or greater.");
                      }
                      callback();
                    }
                  }]
                })(
                  <Input
                    name="folds"
                    onChange={e =>
                      this.props.changeInput(e)
                    }
                  />
                )}
              </Form.Item>
            </Col>
            <Col span={8}>
              <Form.Item
                label="Random seed"
                wrapperCol={{ span: 24 }}>
                {getFieldDecorator("randomSeed", {
                  initialValue: this.props.defaults.randomSeed,
                  rules: [
                    { required: true, message: "Please input random seed!" },

                  ]
                })(
                  <Input
                    name="randomSeed"
                    onChange={e =>
                      this.props.changeInput(e)
                    }
                  />
                )}
              </Form.Item>
            </Col>
            <Col span={8}>
              <Form.Item
                label={(
                  <Tooltip title="The proportion of the dataset that should be included in the test splie">
                    Test size&nbsp;
                  </Tooltip>
                )}
                wrapperCol={{ span: 24 }}>
                {getFieldDecorator("testSize", {
                  initialValue: this.props.defaults.testSize,
                  rules: [
                    { required: true, message: "Please input test size!" },
                    {
                      validator: (rule, value, callback) => {
                        if (value < 0 || value > 1) {
                          callback("Test size must be between zero and one.");
                        }
                        callback();
                      }
                    }
                  ]
                })(
                  <Input
                    name="testSize"
                    onChange={e =>
                      this.props.changeInput(e)
                    }
                  />
                )}
              </Form.Item>
            </Col>
          </Row>
        </Form>
        <Divider dashed />

        <Row type="flex" justify="end">
          <Col>
            <Button
              style={{ marginRight: '15px' }}
              type="default"
              disabled={!this.isValid()}
              onClick={e => this.props.back()}
            >
              <Icon type="left" /> Back
        </Button>
            <Button
              type="primary"
              disabled={!this.isValid()}
              onClick={e => this.props.submit()}
            >
              <Icon type="check" /> Submit
        </Button>
          </Col>
        </Row>

      </div>
    );
  }
}

export const CrossValidationOptionsForm = Form.create()(CrossValidationOptions);
