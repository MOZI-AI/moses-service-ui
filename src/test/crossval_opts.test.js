import React from 'react';
import { mount } from 'enzyme';
import { CrossValidationOptionsForm } from '../crossval_opts';
import renderer from 'react-test-renderer';

const crossValOptions = {
  folds: 0,
  testSize: 3,
  randomSeed: 5
};

describe('<CrossValidationOptionsForm />', () => {
  it('rendered correctly', () => {
    const tree = renderer
      .create(
        <CrossValidationOptionsForm defaults={crossValOptions} show="true" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows error when test size is not between zero and one', () => {
    const wrapper = mount(
      <CrossValidationOptionsForm
        defaults={crossValOptions}
        changeInput={() => {}}
        show="true"
      />
    );
    wrapper
      .find('input[name="testSize"]')
      .simulate('change', { target: { value: 5 } });
    expect(
      wrapper.find('.has-error input[name="testSize"]').exists()
    ).toBeTruthy();
  });

  it('shows error when number of folds is less than one', () => {
    const wrapper = mount(
      <CrossValidationOptionsForm
        defaults={crossValOptions}
        changeInput={() => {}}
        show="true"
      />
    );
    wrapper
      .find('input[name="folds"]')
      .simulate('change', { target: { value: 0 } });
    expect(
      wrapper.find('.has-error input[name="folds"]').exists()
    ).toBeTruthy();
  });
});
