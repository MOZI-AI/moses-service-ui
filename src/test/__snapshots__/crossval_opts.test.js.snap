// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CrossValidationOptionsForm /> rendered correctly 1`] = `
<div
  className="crossValidationOptionsFormWrapper"
>
  <form>
    <div
      className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
    >
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-4-33"
      >
        <div
          aria-describedby={null}
          className="MuiFormControl-root-106 MuiFormControl-marginDense-108 MuiFormControl-fullWidth-109"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Number of folds for cross-validation"
        >
          <label
            className="MuiFormLabel-root-121 MuiInputLabel-root-110 MuiInputLabel-formControl-115 MuiInputLabel-animated-118 MuiInputLabel-marginDense-116 MuiInputLabel-outlined-120"
            data-shrink={false}
          >
            Folds
          </label>
          <div
            className="MuiInputBase-root-141 MuiOutlinedInput-root-128 MuiInputBase-fullWidth-150 MuiInputBase-formControl-142 MuiInputBase-marginDense-148"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-158 MuiOutlinedInput-notchedOutline-135"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-159"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-151 MuiOutlinedInput-input-136 MuiInputBase-inputMarginDense-152 MuiOutlinedInput-inputMarginDense-137"
              defaultValue={1}
              disabled={false}
              name="folds"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Folds"
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-4-33"
      >
        <div
          aria-describedby={null}
          className="MuiFormControl-root-106 MuiFormControl-marginDense-108 MuiFormControl-fullWidth-109"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="Specifies how many times to run MOSES with a random seed value"
        >
          <label
            className="MuiFormLabel-root-121 MuiInputLabel-root-110 MuiInputLabel-formControl-115 MuiInputLabel-animated-118 MuiInputLabel-marginDense-116 MuiInputLabel-outlined-120"
            data-shrink={false}
          >
            Number of seeds
          </label>
          <div
            className="MuiInputBase-root-141 MuiOutlinedInput-root-128 MuiInputBase-fullWidth-150 MuiInputBase-formControl-142 MuiInputBase-marginDense-148"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-158 MuiOutlinedInput-notchedOutline-135"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-159"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-151 MuiOutlinedInput-input-136 MuiInputBase-inputMarginDense-152 MuiOutlinedInput-inputMarginDense-137"
              defaultValue={5}
              disabled={false}
              name="randomSeed"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Number of seeds"
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-2 MuiGrid-grid-xs-4-33"
      >
        <div
          aria-describedby={null}
          className="MuiFormControl-root-106 MuiFormControl-marginDense-108 MuiFormControl-fullWidth-109"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          title="The proportion of the dataset that should be included in the test splie"
        >
          <label
            className="MuiFormLabel-root-121 MuiInputLabel-root-110 MuiInputLabel-formControl-115 MuiInputLabel-animated-118 MuiInputLabel-marginDense-116 MuiInputLabel-outlined-120"
            data-shrink={false}
          >
            Test size
          </label>
          <div
            className="MuiInputBase-root-141 MuiOutlinedInput-root-128 MuiInputBase-fullWidth-150 MuiInputBase-formControl-142 MuiInputBase-marginDense-148"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-158 MuiOutlinedInput-notchedOutline-135"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-159"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input-151 MuiOutlinedInput-input-136 MuiInputBase-inputMarginDense-152 MuiOutlinedInput-inputMarginDense-137"
              defaultValue={0.3}
              disabled={false}
              name="testSize"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Test size"
              required={false}
              type="text"
            />
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
